docs/release-info/release-notes.md

```diff
@@ -24,6 +24,30 @@ title: "Release notes"
 
 <!--Replace {{DRUIDVERSION}} with the correct Druid version.-->
 
+## Druid 29.0.1
+
+Apache Druid 29.0.1 is a patch release that fixes some issues in the Druid 29.0.0 release.
+
+### Bug fixes
+
+* Added type verification for INSERT and REPLACE to validate that strings and string arrays aren't mixed [#15920](https://github.com/apache/druid/pull/15920)
+* [Concurrent replace](https://druid.apache.org/docs/latest/ingestion/concurrent-append-replace) now allows pending Peon segments to be upgraded using the Supervisor [#15995](https://github.com/apache/druid/pull/15995)
+* Changed the `targetDataSource` attribute to return a string containing the name of the datasource. This reverts the breaking change introduced in Druid 29.0.0 for INSERT and REPLACE MSQ queries [#16004](https://github.com/apache/druid/pull/16004) [#16031](https://github.com/apache/druid/pull/16031)
```

Comment:
@ektravel this line should go in the upgrade notes too cause it's an incompatible change.

Reply:
@317brian [upgrade-notes.md](https://github.com/apache/druid/blob/master/docs/release-info/upgrade-notes.md) on the master branch, correct?